Author: Russell Coker <russell@coker.com.au>
Description: Adding volume label and UUID support for linux swap (Closes: #489865).

diff -Naurp file.orig/magic/Magdir/linux file/magic/Magdir/linux
--- file.orig/magic/Magdir/linux	2008-06-04 18:02:25.000000000 +0000
+++ file/magic/Magdir/linux	2008-07-15 14:16:43.000000000 +0000
@@ -56,11 +56,19 @@
 # Linux swap file with swsusp1 image, from Jeff Bailey <jbailey@ubuntu.com>
 4076	string		SWAPSPACE2S1SUSPEND	Linux/i386 swap file (new style) with SWSUSP1 image
 # according to man page of mkswap (8) March 1999
-4086	string		SWAPSPACE2	Linux/i386 swap file (new style)
->0x400	long		x		%d (4K pages)
->0x404	long		x		size %d pages
->>4086	string		SWAPSPACE2	
->>>1052	string		>\0		Label %s
+# volume label and UUID Russell Coker
+# http://etbe.coker.com.au/2008/07/08/label-vs-uuid-vs-device/
+4086	string		SWAPSPACE2	Linux/i386 swap file (new style),
+>0x400	long		x		version %d (4K pages),
+>0x404	long		x		size %d pages,
+>1052	string		\0		no label,
+>1052	string		>\0		LABEL=%s,
+>0x40c	belong		x		UUID=%x
+>0x410	beshort		x		\b-%x
+>0x412	beshort		x		\b-%x
+>0x414	beshort		x		\b-%x
+>0x416	belong		x		\b-%x
+>0x41a	beshort		x		\b%x
 # ECOFF magic for OSF/1 and Linux (only tested under Linux though)
 #
 #	from Erik Troan (ewt@redhat.com) examining od dumps, so this
